staging/src/k8s.io/component-base/metrics/counter_test.go

```diff
@@ -285,3 +294,119 @@ func TestCounterWithLabelValueAllowList(t *testing.T) {
 		})
 	}
 }
+
+func TestCounterWithExemplar(t *testing.T) {
+	// Set exemplar.
+	traceID := trace.TraceID{1, 2, 3}
+	spanID := trace.SpanID{4, 5, 6}
+	ctxForSpanCtx := trace.ContextWithSpanContext(context.Background(), trace.NewSpanContext(trace.SpanContextConfig{
+		TraceID: traceID,
+		SpanID:  spanID,
+	}))
+	traceIDStr := traceID.String()
+	spanIDStr := spanID.String()
+	toAdd := float64(41)
+
+	// Create contextual counter.
+	portStr := ":" + strconv.Itoa(8090)
+	counter := NewCounter(&CounterOpts{
+		Name: "metric_exemplar_test",
+		Help: "helpless",
+	})
+	_ = counter.WithContext(ctxForSpanCtx)
+
+	// Register counter.
+	registry := newKubeRegistry(apimachineryversion.Info{
+		Major:      "1",
+		Minor:      "15",
+		GitVersion: "v1.15.0-alpha-1.12345",
+	})
+	registry.MustRegister(counter)
+
+	// Call underlying exemplar methods.
+	go func() {
+		counter.Add(toAdd)
+		counter.Inc()
+
+		// Synthetic latency.
+		time.Sleep(600 * time.Millisecond)
+	}()
+
+	// Verify value.
+	err := wait.PollUntilContextTimeout(context.Background(), 100*time.Millisecond, time.Second, false, func(ctx context.Context) (done bool, err error) {
+		mfs, err := registry.Gather()
+		if err != nil {
+			return false, err
+		}
+		gotValue := mfs[0].GetMetric()[0].GetCounter().GetValue()
+		wantValue := toAdd + 1
+		if gotValue != wantValue {
+			return false, nil
+		}
+		return true, nil
+	})
+	if err != nil {
+		t.Fatalf("failed to verify value: %v", err)
+	}
+
+	// Start an open-metrics-compatible server.
+	http.Handle(
+		"/metrics", promhttp.HandlerFor(
+			registry,
+			promhttp.HandlerOpts{
+				EnableOpenMetrics: true,
```

Comment:
we want to request the metrics in proto in prometheus exposition format.

Reply:
[from the call] Exemplar support for `ProtoText` is not yet implemented, and since `FmtOpenMetrics` format is [nearing archive](https://groups.google.com/g/openmetrics/c/czniCbgA2j8/m/3ke0Syk8AAAJ?pli=1) the best way forward would be to implement that support for [`ProtoFmt`](https://github.com/prometheus/common/blob/main/expfmt/expfmt.go#L25) (and some or all other `content-type`s that are based on that format).